src/acme/ui/CustomerListUI.java

```diff
@@ -0,0 +1,319 @@
+package acme.ui;
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Font;
+import java.awt.Panel;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.ResultSet;
+
+import javax.swing.DefaultCellEditor;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableCellRenderer;
+import javax.swing.table.TableColumn;
+
+import acme.pd.User;
+import acme.pd.Customer;
+import acme.pd.MapIntersection;
+import acme.pd.Road;
+
+public class CustomerListUI extends AcmeBaseJPanel {
+  
+  public CustomerListUI() {
+    super();
+  }
+  
+  public void buildPanel() {
+  
+    setLayout(null);
+    
+    //title label
+    JLabel lblCustomers = new JLabel("Customers");
+    lblCustomers.setFont(new Font("Lucida Grande", Font.PLAIN, 24));
+    lblCustomers.setBounds(20, 20, 200, 27);
+    add(lblCustomers);
+    
+    //my test data
```

Comment:
Should be able to grab some customers from the company, I think there are a few being created at startup right now

Reply:
I tried to grab the data I couldn't get any. But it was useful, because I reworked the table to go with an arbitrary size of data and got the edit customer button working fully.